lib/service/storage/virtualIndex.ts

```diff
@@ -0,0 +1,200 @@
+import { Service } from "../service";
+
+export class VirtualIndex extends Service {
+  public static createEvent = "core:storage:virtual-index:cache:create";
+  public static deleteEvent = "core:storage:virtual-index:cache:delete";
+
+  public static createVirtualIndexEvent =
+    "core:storage:virtual-index:create:after";
+
+  public static deleteVirtualIndexEvent =
+    "core:storage:virtual-index:delete:after";
+
+  constructor() {
+    super("VirtualIndex", global.kuzzle.config.services.storageEngine);
+  }
+
+  public virtualIndexMap: Map<string, string> = new Map(); //Key : virtual index, value : real index
+
+  async init(): Promise<void> {
+    await this.buildCollection();
+    await this.initVirtualIndexList();
+
+    global.kuzzle.onAsk(VirtualIndex.createEvent, (physical, virtual) =>
+      this.addInSoftIndexMap(physical, virtual)
+    );
+
+    global.kuzzle.onAsk(VirtualIndex.deleteEvent, (info) =>
+      this.removeInSoftIndexMap(info)
+    );
+  }
+
+  addInSoftIndexMap(physical, virtual) {
```

Comment:
Nitpicking: here the naming should be `addInVirtualIndexMap` instead

Reply:
Done